providers/oidc.go

```diff
@@ -40,6 +41,7 @@ func (p *OIDCProvider) Redeem(ctx context.Context, redirectURL, code string) (*s
 			TokenURL: p.RedeemURL.String(),
 		},
 		RedirectURL: redirectURL,
+		Scopes:      strings.Split(p.Scope, ","),
```

Comment:
Scopes are handled here on the authorization endpoint stage: https://github.com/oauth2-proxy/oauth2-proxy/blob/f3209a40e16defdab442ef87765e837d90ec4f9a/providers/util.go#L50

Do you have documentation for your IdP where they are expecting scope in the token endpoint instead of the authorization endpoint?

Reply:
Your comment totally makes sense. Let me double check from my side and if I feel this change is not needed I'll close it 👍

Reply:
My understanding was most/all OIDC providers would take the scope into effect when the make your session on their end during the `authorization_endpoint` request that generates the authentication code that the redeem token endpoint uses. Then the scope is properly set in later stages after you redeem that code in both you ID Token claims and any calls to the userinfo endpoint with you Access Token. And scope is remembered by them later when you use the refresh token on then token endpoint to refresh everything.

I wonder if `Scopes` on the `oauth2.Config` for the token endpoint is just for OAuth providers that allow overriding the scopes on secondary token/refresh requests? I don't know the history of that option.

Reply:
My fault. I checked again without this PR and it works without a problem.

Probably I wasn't specifying `p.Scope` before testing, added it in the wrong place because I thought it was missing for OIDC and while doing so I might have added it in my config: thus making it work.

Let's discard this PR. Sorry for wasting your time!